src/common/transformations/include/ov_ops/fully_connected.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+#include "openvino/op/op.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace op {
+namespace internal {
+
+class TRANSFORMATIONS_API FullyConnected : public ov::op::Op {
+public:
+    OPENVINO_OP("FullyConnected", "ie_internal_opset");
+
+    FullyConnected() = default;
+
+    FullyConnected(const OutputVector& arguments, const ov::element::Type output_type = ov::element::undefined);
+
+    FullyConnected(const ov::Output<Node>& A,
+                   const ov::Output<Node>& B,
+                   const ov::Output<Node>& bias,
+                   const ov::element::Type output_type = ov::element::undefined);
+
+    bool visit_attributes(ov::AttributeVisitor& visitor) override;
+
+    void validate_and_infer_types() override;
+
+    std::shared_ptr<Node> clone_with_new_inputs(const ov::OutputVector& new_args) const override;
+
+    virtual std::shared_ptr<Node> fuse_bias(const ov::Output<Node>& bias) const;
```

Comment:
Do we need a method for that, especially virtual? I think bias fusion can be simply done via `clone_with_new_inputs` for any FC type:
```cpp

        ov::OutputVector inputs;
        for (size_t i = 0; i < fc->get_input_size(); i++) {
            inputs.push_back(fc->get_input_source_output(i)); // would be nice to have a such helper in ov::Node
        }
        inputs[2] = bias;
        fc->clone_with_new_inputs(inputs);
```

Reply:
Agree, virtual method is not necessary anymore.
This was implemented before I have added a constructor which accepts OutputVector.
But maybe we still could benefit from some class static function.